s3fs/mapwrap.py

```diff
@@ -0,0 +1,73 @@
+
+from collections import MutableMapping
+import os
+
+class MapWrap(MutableMapping):
+    """Wrap an S3FileSystem as a mutable wrapping.
+
+    The keys of the mapping become files under the given root, and the
+    values (which must be bytes) the contents of those files.
+
+    Usage
+    -----
```

Comment:
Common header is `Examples`.  Documentation systems will pick up this header and not custom ones like `Usage`.  Also suggest adding a `Parameters` section.

Additionally, for users unfamiliar with the s3fs project we might want to point them to how to build the `s3` object, either by adding a `See Also` link or by including the creation of an `S3FileSystem` object in the example.